ludwig/utils/fs_utils.py

```diff
@@ -257,6 +272,7 @@ def put_fn():
         yield url, None
 
 
+# TODO(shreya): Add retry for open context.
```

Comment:
@arnavgarg1 it's not possible to execute the code in a context manager multiple times. More details [here](https://stackoverflow.com/a/36741101).

My plan to add retries for this are to change the context so that `open_file` calls `fs.open` on entering the context (I'll have to double check if an `fs.close` is needed on exit), and then wrap the `open` (and `close`) calls in `default_retry_call` wrappers.

Reply:
That makes sense!